Datastore/src/Operation.php

```diff
@@ -510,7 +592,31 @@ public function runQuery(QueryInterface $query, array $options = [])
                 $runQueryObj->queryKey() => $requestQueryArr,
             ] + $this->readOptions($options) + $options;
 
-            $res = $this->connection->runQuery($request);
+            list($data, $optionalArgs) = $this->splitOptionalArgs($req, [
+                'className',
+                'namespaceId',
+                'readTime',
+                'readConsistency',
+                'transaction'
+            ]);
+            $data = $this->convertDataToProtos($data, [
+                'partitionId' => PartitionId::class,
+                'readOptions' => ReadOptions::class,
+                'explainOptions' => ExplainOptions::class
+            ]);
+            if (isset($data['query'])) {
+                $data['query'] = $this->parseQuery($data['query']);
```

Comment:
Was this processing happening in the connection layer before?
A few comments will be helpful here.

Can even consider extracting this into a private method, specially if it's being repeated.

Reply:
Yes this processing happened in at the Grpc connection [layer](https://github.com/googleapis/google-cloud-php/blob/9ad2e3d7fd8358ef7f9d8ca9c6c4db68dbcadb78/Datastore/src/Connection/Grpc.php#L265-L275).

Since it isn't repeating, hence extracting doesn't makes any sense here.

Reply:
Done in https://github.com/yash30201/google-cloud-php/pull/6